hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/scm/storage/ContainerProtocolCalls.java

```diff
@@ -292,6 +292,34 @@ public static XceiverClientReply putBlockAsync(XceiverClientSpi xceiverClient,
     return xceiverClient.sendCommandAsync(request);
   }
 
+  static XceiverClientReply putBlockAsync(XceiverClientSpi xceiverClient,
+                                                 BlockData containerBlockData, boolean eof,
+                                                 String tokenString)
+      throws IOException, InterruptedException, ExecutionException {
+    final ContainerCommandRequestProto request = getPutBlockRequest(
+        xceiverClient.getPipeline(), containerBlockData, eof, tokenString);
+    return xceiverClient.sendCommandAsync(request);
+  }
+
+  public static ContainerCommandRequestProto getPutBlockRequest(
+      Pipeline pipeline, BlockData containerBlockData, boolean eof,
+      String tokenString) throws IOException {
+    PutBlockRequestProto.Builder createBlockRequest =
+        PutBlockRequestProto.newBuilder()
+            .setBlockData(containerBlockData)
+            .setEof(eof);
+    final String id = pipeline.getFirstNode().getUuidString();
+    ContainerCommandRequestProto.Builder builder =
+        ContainerCommandRequestProto.newBuilder().setCmdType(Type.PutBlock)
+            .setContainerID(containerBlockData.getBlockID().getContainerID())
+            .setDatanodeUuid(id)
+            .setPutBlock(createBlockRequest);
```

Comment:
Duplicating this part can be avoided by calling the new method from the existing one:

```java
  public static ContainerCommandRequestProto getPutBlockRequest(
      Pipeline pipeline, BlockData containerBlockData, boolean eof,
      Token<? extends TokenIdentifier> token) throws IOException {
    final String tokenString = token != null ? token.encodeToUrlString() : null;
    return getPutBlockRequest(pipeline, containerBlockData, eof, tokenString);
  }
```